pandas-stubs/io/parsers/readers.pyi

```diff
@@ -48,7 +48,10 @@ def read_csv(
     | None = ...,
     dtype: DtypeArg | defaultdict | None = ...,
     engine: CSVEngine | None = ...,
-    converters: dict[int | str, Callable[[str], Any]] | None = ...,
+    converters: dict[int | str, Callable[[str], Any]]
+    | dict[int, Callable[[str], Any]]
+    | dict[str, Callable[[str], Any]]
```

Comment:
Can you try `Mapping[int | str, Callable[[str], Any]]` instead of the union?

Reply:
I tried that first. But it seems `Mapping` isn't covariant over the key, only the value: https://github.com/python/typing/issues/445

Reply:
But mapping may work for that the `Type[str]` issue since that is the value...will try that

I also wonder if creating a covariant `TypeVar` and then a `Mapping[T_co, ...]` will work...I will also try that.